torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -147,6 +148,104 @@ def post_compile(
+        self.replay_views = config.view_replay_for_aliased_outputs
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = orig_inputs[self.base_idx]
+        return gen_alias_from_base(
+            aliased_base_tensor,
+            self.unwrap_out(out),
+            self.requires_grad,
+            self.functional_tensor,
+            replay_views=self.replay_views,
+        )
+
+
+class IsInputHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        num_tokens = len(runtime_metadata.tokens)
+        self.base_idx = info.base_idx + num_tokens
+        self.unwrap_out = _unwrap_tensoralias if trace_joint else _identity
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = orig_inputs[self.base_idx]
+        return aliased_base_tensor
+
+
+class AliasOfIntermediateHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        if info.output_type in (
+            OutputType.alias_of_intermediate,
+            OutputType.alias_of_intermediate_save_as_output,
+        ):
+            num_user_outputs = len(runtime_metadata.output_info)
+            self.base_idx = info.base_idx + num_user_outputs
+        else:
+            self.base_idx = info.base_idx
+
+        self.unwrap_out = _unwrap_tensoralias if trace_joint else _identity
+        self.requires_grad = info.requires_grad
+        self.functional_tensor = info.functional_tensor
+        self.replay_views = config.view_replay_for_aliased_outputs
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = fw_outs[self.base_idx]
+        return gen_alias_from_base(
+            aliased_base_tensor,
+            self.unwrap_out(out),
+            self.requires_grad,
+            self.functional_tensor,
+            replay_views=self.replay_views,
+        )
+
+
+def make_output_handler(info, runtime_metadata, trace_joint):
+    if info.output_type in (
+        OutputType.non_alias,
+        OutputType.unsafe_view_alias,
+        OutputType.custom_function_view,
+    ):
+        return NoopAliasHandler(info, runtime_metadata, trace_joint)
+    elif info.output_type == OutputType.alias_of_input:
+        return AliasOfInputHandler(info, runtime_metadata, trace_joint)
+    elif info.output_type == OutputType.is_input:
+        return IsInputHandler(info, runtime_metadata, trace_joint)
+    elif info.output_type in (
+        OutputType.alias_of_intermediate,
+        OutputType.alias_of_intermediate_save_as_output,
+        OutputType.alias_of_intermediate_base_is_user_output,
+    ):
+        return AliasOfIntermediateHandler(info, runtime_metadata, trace_joint)
+    # TODO: handle the custom autograd function case here.
+    # We need a way to check whether a tensor came from a custom autograd fn from python,
+    # AND a way to replay that custom view fn.
+    raise AssertionError(f"Unhandled output type {info.output_type}")
+
+
 def _create_runtime_wrapper(
```

Comment:
When I did this refactor I didn't have as much context on runtime wrapper so I just copied the existing helper function, but I actually think the correct thing to do here is to minimize the work _create_runtime_wrapper does in this function and instead have a pre_compile function that handles it. It would make it more clear that the win is moving this processing from runtime to compile time.

Reply:
I don't think it's that confusing tbh. If it's inside the `runtime_wrapper` closure then it's happening at runtime and vice versa.